src/test/Fake.Core.UnitTests/Fake.Core.Target.fs

```diff
@@ -455,4 +455,133 @@ let tests =
       let context = run "SimpleTest"
       Expect.equal "Expected failure" true context.HasError
       Expect.equal "Expected context to contain both targets" 2 context.PreviousTargets.Length  // second one as "skipped"
+
+    targetTestCase "Not activated final target does not run" <| fun _ ->
+      Target.create "a" ignore
+      Target.create "b" ignore
+      Target.create "c" ignore
+      "a" ==> "b" ==> "c" |> ignore
+      let mutable finalTargetResult = 0
+      Target.createFinal "Final" (fun _ -> finalTargetResult <- 1)
+            
+      let context = run "c"
```

Comment:
Maybe we should let some tests run with a parallel version of `run`

Reply:
I was actually going to add tests for the cancellation, but i am not sure how to do that. It would probaby have to run in another Process? Ans since i did not really manage to do that i at least added some tests for the normal final-targets :/. 
CoreFx has some [here](https://github.com/dotnet/corefx/blob/master/src/System.Console/tests/CancelKeyPress.Unix.cs). But that is quite a lot of boilerblate to add.

Reply:
I think cancellation is fine, I meant is just letting it run through the parallel code paths and see if the result still is fine (yes in fact that might mean that stuff is our of order if dependencies allow it but we can still do basic checks)